File: Randomizer-MacOS/Randomizer-MacOS/ViewController.m
//
//  ViewController.m
//  Randomizer-MacOS
//
//  Created by Leo Dav on 08/05/22.
//

#import "ViewController.h"

@interface ViewController()

@property (weak) IBOutlet NSNumberFormatter* formatter;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.formatter.allowsFloats = NO;
    self.maxBound = 1;
    self.formatter.nilSymbol = @"1";
}

- (IBAction)generate:(NSButton *)sender {
    unsigned int generated = arc4random_uniform(self.maxBound) + 1;
    self.labelGenerated.integerValue = generated;
}


@end

File: Randomizer-MacOS/Randomizer-MacOS/ViewController.swift
//
//  ViewController.swift
//  Randomizer-MacOS
//
//  Created by Leo Dav on 23/04/2022.
//
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet weak var textBox: NSTextField!
    @IBOutlet weak var resultLabel: NSTextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override var representedObject: Any? {
        didSet {
            // Update the view, if already loaded.
        }
    }

    @IBAction func onGenerate(_ sender: NSButton) {
        print("You Touched the Button")
    }
    
    @IBAction func onValueChanged(_ sender: NSStepper) {
        textBox.integerValue = sender.integerValue
    }
}
